packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -216,40 +259,55 @@ export class RichText extends Component {
 	}
 
 	onFormatChange( record, doUpdateChild ) {
-		let newContent;
-		// valueToFormat might throw when converting the record to a tree structure
-		// let's ignore the event for now and force a render update so we're still in sync
-		try {
-			newContent = this.valueToFormat( record );
-		} catch ( error ) {
-			// eslint-disable-next-line no-console
-			console.log( error );
-		}
-		this.setState( {
-			formatPlaceholder: record.formatPlaceholder,
+		const { start, end, activeFormats = [] } = record;
+		const changeHandlers = pickBy( this.props, ( v, key ) =>
+			key.startsWith( 'format_on_change_functions_' )
+		);
+
+		Object.values( changeHandlers ).forEach( ( changeHandler ) => {
+			changeHandler( record.formats, record.text );
 		} );
-		if ( newContent && newContent !== this.props.value ) {
-			this.props.onChange( newContent );
-			if ( record.needsSelectionUpdate && record.start && record.end && doUpdateChild ) {
-				this.forceSelectionUpdate( record.start, record.end );
-			}
-		} else {
-			if ( doUpdateChild ) {
-				this.lastEventCount = undefined;
-			} else {
-				// make sure the component rerenders without refreshing the text on gutenberg
-				// (this can trigger other events that might update the active formats on aztec)
-				this.lastEventCount = 0;
-			}
-			this.forceUpdate();
+
+		this.value = this.valueToFormat( record );
+		this.props.onChange( this.value );
+		this.setState( { activeFormats } );
+		this.props.onSelectionChange( start, end );
+		this.selectionStart = start;
+		this.selectionEnd = end;
+
+		this.onCreateUndoLevel();
+
+		// ////////////
+
+		// let newContent = this.value;
+		// if ( newContent && newContent !== this.props.value ) {
+		// 	///
+		// } else {
+		// 	if ( doUpdateChild ) {
+		// 		this.lastEventCount = undefined;
```

Comment:
@hypest - This is the source of the format issue on iOS. If eventCount is other than undefined, Aztec-iOS will not update its content, so we need to undefine it at some point.

Reply:
I'm not sure what was the reason to do this change, but if it's working properly on Android, maybe we can modify the native iOS Aztec wrapper to work as the Android one does?

Reply:
Chatted over Slack and I think the culprit is to be found in the `shouldComponentUpdate` function? The assumption we can try to align to is that a "normal" prop change (initiated by https://github.com/WordPress/gutenberg/pull/15392/files/380660e7d751834ac0de4eb7f207d0c0f89fa863#diff-4828a21853e899e5a36faecfa96d83e8R272) should be enough to cause the code to try and force the changes to the Aztecs.